### DISCUSSION TITLE
Elastic agent: An existing connection was forcibly closed by the remote host

### DISCUSSION THREAD

**Discussion by CharlieMK** [Security-Onion-Solutions/securityonion#11142] [2.4] [answered]
### Version

2.4.10

### Installation Method

Security Onion ISO image

### Description

configuration

### Installation Type

Standalone

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### Network Traffic Collection

tap

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

I installed the hotfix for 2.4.10 today, and my other issue was resolved: https://github.com/Security-Onion-Solutions/securityonion/discussions/11109. 

I then installed the elastic agent for a windows machine, but I'm getting errors as the following in the agent info dashboard:
`failed to publish events: write tcp agentIP:27601->SOIP:5055: wsasend: An existing connection was forcibly closed by the remote host.`

I set the 'elastic_agent_endpoint' setting to the CIDR of the agentIP (192.168.X.0/24) and waited half a day. The issue is not resolved. 

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by defensivedepth** [comment]
You used the installer from the Security Onion Console --> Downloads page?

What do the logs show on the Windows host itself, under program files / elastic / agent ?

**Comment by CharlieMK** [comment]
> You used the installer from the Security Onion Console --> Downloads page?

Yes, that is correct. 

> What do the logs show on the Windows host itself, under program files / elastic / agent ?

There are quite some logs, an extract: 

```
{"log.level":"error","@timestamp":"2023-08-23T04:21:23.777Z","message":"Failed to publish events caused by: write tcp agentIP:53379->SOIP:5055: wsasend: An existing connection was forcibly closed by the remote host.","component":{"binary":"filebeat","dataset":"elastic_agent.filebeat","id":"winlog-default","type":"winlog"},"log":{"source":"winlog-default"},"log.logger":"logstash","log.origin":{"file.line":280,"file.name":"logstash/async.go"},"service.name":"filebeat","ecs.version":"1.6.0","ecs.version":"1.6.0"}
{"log.level":"error","@timestamp":"2023-08-23T04:21:25.050Z","message":"failed to publish events: write tcp agentIP:53379->SOIP:5055: wsasend: An existing connection was forcibly closed by the remote host.","component":{"binary":"filebeat","dataset":"elastic_agent.filebeat","id":"winlog-default","type":"winlog"},"log":{"source":"winlog-default"},"log.logger":"publisher_pipeline_output","log.origin":{"file.line":174,"file.name":"pipeline/client_worker.go"},"service.name":"filebeat","ecs.version":"1.6.0","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2023-08-23T04:21:25.050Z","message":"Connecting to failover(backoff(async(tcp://securityonion.domain.com:5055)),backoff(async(tcp://securityonion:5055)))","component":{"binary":"filebeat","dataset":"elastic_agent.filebeat","id":"winlog-default","type":"winlog"},"log":{"source":"winlog-default"},"log.logger":"publisher_pipeline_output","log.origin":{"file.line":137,"file.name":"pipeline/client_worker.go"},"service.name":"filebeat","ecs.version":"1.6.0","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2023-08-23T04:21:25.139Z","message":"Connection to failover(backoff(async(tcp://securityonion.domain.com:5055)),backoff(async(tcp://securityonion:5055))) established","component":{"binary":"filebeat","dataset":"elastic_agent.filebeat","id":"winlog-default","type":"winlog"},"log":{"source":"winlog-default"},"service.name":"filebeat","ecs.version":"1.6.0","log.logger":"publisher_pipeline_output","log.origin":{"file.line":145,"file.name":"pipeline/client_worker.go"},"ecs.version":"1.6.0"}
```
Next a whole bunch of 'Non-zero metrics in the last 30s' and then it repeats with the closing error.

**Reply by defensivedepth** [reply]
Is the Agent healthy in Fleet?

Is the Agent show any errors when you run the following on the windows system:

C:\program files\elastic\agent\ 

`.\elastic-agent.exe status`

**Reply by CharlieMK** [reply]
The agent shows up as 'unhealthy' in fleet. 

output of elastic-agent.exe status:
```
State: DEGRADED
Message: 1 or more components/units in a failed state
Fleet State: HEALTHY
Fleet Message: Connected
Components:
  *             (FAILED)
                input not supported
  * osquery     (HEALTHY)
                Healthy: communicating with pid '10000'
  * log         (HEALTHY)
                Healthy: communicating with pid '10352'
  * winlog      (HEALTHY)
                Healthy: communicating with pid '10520'
  * filestream  (HEALTHY)
                Healthy: communicating with pid '10748'
```

**Reply by CharlieMK** [reply, marked as answer]
Taking inspiration from https://github.com/Security-Onion-Solutions/securityonion/discussions/11148, I deleted the elastic-defend integration from the endpoints-initial policy and re-added it with 'data collection' setting. 
The issue seems to be resolved.

**Comment by defensivedepth** [comment]
@CharlieMK We have confirmed this as an issue. Looks like you already got it fixed!
